v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSourceDbIT.java

```diff
@@ -0,0 +1,235 @@
+    // Create MySQL Resource manager
+    jdbcResourceManager = CustomMySQLResourceManager.builder(testName).build();
+
+    // Arrange MySQL-compatible schema
+    HashMap<String, String> columns = new HashMap<>();
+    columns.put("id", "INT NOT NULL");
+    columns.put("name", "VARCHAR(25)");
+    JDBCResourceManager.JDBCSchema schema = new JDBCResourceManager.JDBCSchema(columns, "id");
+
+    // Create JDBC table
+    jdbcResourceManager.createTable(TABLE, schema);
+
+    MySQLSource mySQLSource =
+        MySQLSource.builder(
+                jdbcResourceManager.getHost(),
+                jdbcResourceManager.getUsername(),
+                jdbcResourceManager.getPassword(),
+                jdbcResourceManager.getPort())
+            .build();
+
+    spannerResourceManager = createSpannerDatabase();
+    spannerMetadataResourceManager = createSpannerMetadataDatabase();
+
+    // Run a simple IT
+    simpleGcsToSourceDbTest(
+        testName,
+        TABLE,
+        mySQLSource,
+        DatastreamResourceManager.DestinationOutputFormat.AVRO_FILE_FORMAT,
+        b -> b.addParameter("inputFileFormat", "avro"));
+  }
+
+  private SpannerResourceManager createSpannerDatabase() throws IOException {
+    SpannerResourceManager spannerResourceManager =
+        SpannerResourceManager.builder("rr-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String ddl =
+        String.join(
+            " ",
+            Resources.readLines(
+                Resources.getResource(SPANNER_DDL_RESOURCE), StandardCharsets.UTF_8));
+    ddl = ddl.trim();
+    String[] ddls = ddl.split(";");
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    return spannerResourceManager;
+  }
+
+  private SpannerResourceManager createSpannerMetadataDatabase() throws IOException {
+    SpannerResourceManager spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    return spannerMetadataResourceManager;
+  }
+
+  private void simpleGcsToSourceDbTest(
+      String testName,
+      String tableName,
+      JDBCSource jdbcSource,
+      DatastreamResourceManager.DestinationOutputFormat fileFormat,
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder)
+      throws IOException, InterruptedException {
+
+    createAndUploadShardConfigToGcs();
+    gcsClient.uploadArtifact(
+        "input/session.json", Resources.getResource(SESSION_FILE_RESOURSE).getPath());
+    launchWriterDataflowJob();
+
+    // Write events to GCS
+    gcsClient.uploadArtifact(
```

Comment:
The file name is expected to be in a specific naming format. Please refer the GCSReader.